cpp/Platform.Data.Doublets/ILinksExtensions.h

```diff
@@ -396,24 +396,24 @@ namespace Platform::Data::Doublets
             }
         }
 
-    //    template<typename TStorage>
-    //    static typename TStorage::LinkAddressType CountUsages(TStorage& storage, typename TStorage::LinkAddressType linkAddress)
-    //    {
-    //        auto constants = storage.Constants;
-    //        auto values = GetLink(storage, linkAddress);
-    //        typename TStorage::LinkAddressType usagesAsSource = Count(storage)(Link(constants.Any, linkAddress, constants.Any));
-    //        if ( link == GetSource(storage, values))
-    //        {
-    //            usagesAsSource = usagesAsSource - 1;
-    //        }
-    //        typename TStorage::LinkAddressType usagesAsTarget = Count(storage)(Link(constants.Any, constants.Any, linkAddress));
-    //        if ( link == GetTarget(storage, values))
-    //        {
-    //            usagesAsTarget = usagesAsTarget - 1;
-    //        }
-    //        return usagesAsSource + usagesAsTarget;
-    //    }
-    //
+        template<typename TStorage>
+        static typename TStorage::LinkAddressType CountUsages(TStorage& storage, typename TStorage::LinkAddressType linkAddress)
+        {
+            auto constants = storage.Constants;
+            auto values = GetLink(storage, linkAddress);
+            typename TStorage::LinkAddressType usagesAsSource = Count(storage)(Link(constants.Any, linkAddress, constants.Any));
```

Comment:
`storage` must be passed to `Count` before restriction (Link). By the way, use `typename TStorage::LinkType` instead of `Link`